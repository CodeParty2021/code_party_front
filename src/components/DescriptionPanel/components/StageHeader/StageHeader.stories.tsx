import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { StageHeader } from "./StageHeader";

export default {
  title: "components/DescriptionPanel/StageHeader",
  component: StageHeader,
} as ComponentMeta<typeof StageHeader>;

const Template: ComponentStory<typeof StageHeader & { children: any }> = (
  args
) => <StageHeader {...args} />;

export const Default = Template.bind({});
Default.args = {
  stage: 1,
  completeStep: 0,
  maxStep: 3,
  workingStep: 1,
};
