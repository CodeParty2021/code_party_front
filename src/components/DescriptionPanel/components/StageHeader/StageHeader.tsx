import { MissionNumber } from "components/MissionNumber/MissionNumber";
import React from "react";
import {
  GoalBanner,
  Progress,
  ProgressBlue,
  ProgressGray,
  RightBox,
  StageHeaderStyle,
  StageHeaderStyleProps,
  TextProgress,
  Title,
} from "./StageHeaderStyle";

type Props = StageHeaderStyleProps & {
  stage: number;
  workingStep: number;
  completeStep: number;
  maxStep: number;
};
const progress = (step: number, maxstep: number) =>
  (step / maxstep) * 100 + "%";
export const StageHeader: React.FC<Props> = ({
  stage,
  completeStep,
  workingStep,
  maxStep,
  ...styleProps
}) => {
  return (
    <StageHeaderStyle {...styleProps}>
      <MissionNumber number={stage} />
      <RightBox>
        <Title>まっすぐ歩くロボット</Title>
        <Progress>
          <ProgressGray>
            <ProgressBlue progress={progress(completeStep, maxStep)} />
          </ProgressGray>
          <GoalBanner fill="#6675FC" />
          <TextProgress>
            {workingStep}/{maxStep}
          </TextProgress>
        </Progress>
      </RightBox>
    </StageHeaderStyle>
  );
};
