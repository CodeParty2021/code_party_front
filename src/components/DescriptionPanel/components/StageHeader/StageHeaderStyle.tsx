import { Banner } from "components/icons";
import styled from "styled-components";
import { FlexGap } from "styles/FlexGap/FlexGap";
export type StageHeaderStyleProps = {};

export const StageHeaderStyle = styled.div`
  display: flex;
  ${FlexGap({ gap: "16px", direction: "row" })}
  padding-left:69px;
  width: 100%;
`;
export const RightBox = styled.div`
  display: flex;
  flex-direction: column;
  ${FlexGap({ gap: "8px", direction: "column" })}
  width:100%;
`;
export const Title = styled.div`
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 700;
  font-size: 30px;
  line-height: 43px;
  font-feature-settings: "palt" on;

  /* GRAY_100 */

  color: #242a3d;

  transform: matrix(1, 0, -0.08, 1, 0, 0);
`;
export const Progress = styled.div`
  position: relative;
  height: 8px;
  margin-right: 32px;
`;
export const ProgressGray = styled.div`
  border-radius: 4px;
  width: 100%;
  height: 100%;
  background-color: #d7dae4;
`;
export const ProgressBlue = styled.div<{ progress: string }>`
  ${({ progress }) => `width: ${progress}`};
  height: 8px;
  border-radius: 4px;
  background-color: #6675fc;
  transition: 0.5s;
`;

export const GoalBanner = styled(Banner)`
  position: absolute;
  top: 0;
  right: 0;
  transform: translate(100%, -100%);
`;

export const TextProgress = styled.div`
  position: absolute;
  top: 0;
  right: 8px;
  transform: translate(0, -100%);
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 700;
  font-size: 18px;
  line-height: 26px;
  /* identical to box height */

  font-feature-settings: "palt" on;

  /* GRAY_100 */

  color: #242a3d;
`;
