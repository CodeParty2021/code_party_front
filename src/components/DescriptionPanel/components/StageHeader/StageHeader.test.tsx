import React from "react";
import { StageHeader } from "./StageHeader";
import "jest-styled-components";
import { render } from "enzyme";

describe("<StageHeader />", () => {
  it("snapshot test", () => {
    const wrapper = render(
      <StageHeader
        {...{ stage: 1, completeStep: 0, workingStep: 1, maxStep: 3 }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
