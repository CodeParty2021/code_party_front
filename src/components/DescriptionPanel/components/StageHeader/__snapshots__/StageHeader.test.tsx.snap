// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StageHeader /> snapshot test 1`] = `
.c1 {
  font-family: "851Gkktt";
  font-size: 30px;
  color: #F5F6F8;
  overflow: hidden;
  width: 50px;
  height: 59px;
  position: relative;
}

.c1 > span {
  position: absolute;
  left: 50%;
  top: 12px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c2 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  fill: #2A2C33;
}

.c8 {
  display: inline-block;
  width: 32px;
  height: 32px;
  fill: #6675FC;
}

.c8 > span,
.c8 > div,
.c8 > svg {
  display: block;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  padding-left: 69px;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
  width: 100%;
}

.c4 {
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 700;
  font-size: 30px;
  line-height: 43px;
  font-feature-settings: "palt" on;
  color: #242A3D;
  -webkit-transform: matrix(1,0,-0.08,1,0,0);
  -ms-transform: matrix(1,0,-0.08,1,0,0);
  transform: matrix(1,0,-0.08,1,0,0);
}

.c5 {
  position: relative;
  height: 8px;
  margin-right: 32px;
}

.c6 {
  border-radius: 4px;
  width: 100%;
  height: 100%;
  background-color: #D7DAE4;
}

.c7 {
  width: 0%;
  height: 8px;
  border-radius: 4px;
  background-color: #6675fc;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c9 {
  position: absolute;
  top: 0;
  right: 0;
  -webkit-transform: translate(100%,-100%);
  -ms-transform: translate(100%,-100%);
  transform: translate(100%,-100%);
}

.c10 {
  position: absolute;
  top: 0;
  right: 8px;
  -webkit-transform: translate(0,-100%);
  -ms-transform: translate(0,-100%);
  transform: translate(0,-100%);
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 700;
  font-size: 18px;
  line-height: 26px;
  font-feature-settings: "palt" on;
  color: #242A3D;
}

@supports not (gap:16px) {
  .c0 > *:not(:first-child) {
    margin-left: 16px;
  }
}

@supports not (gap:8px) {
  .c3 > *:not(:first-child) {
    margin-top: 8px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      class="c2"
      xlink="http://www.w3.org/1999/xlink"
      xmlns="http://www.w3.org/2000/svg"
    />
    <span>
      1
    </span>
  </div>
  <div
    class="c3"
  >
    <div
      class="c4"
    >
      まっすぐ歩くロボット
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        />
      </div>
      <span
        class="c8 c9"
        display="inline-block"
        fill="#6675FC"
      />
      <div
        class="c10"
      >
        1/3
      </div>
    </div>
  </div>
</div>
`;
